src/raft/raft.rs

```diff
@@ -304,6 +304,11 @@ impl<T: Storage> Raft<T> {
         hs
     }
 
+    pub fn in_lease(&self, safe_distance: usize) -> bool {
+        self.state == StateRole::Leader && self.check_quorum &&
+        self.election_elapsed + safe_distance <= self.election_timeout
```

Comment:
the election_elapsed can only be reset after reach election_timeout, assume election_timeout is 20 ticks, so if the leader is already leader, we may still use raft read in [15, 20] ticks? 
only using check_quorum and state check is enough ? /cc @swingbach